### PR TITLE
New Wallet Service + Commit on Service layer

### PR DESCRIPTION
**Pull request by JadeArkadian**
## Summary by CodeRabbit

* **New Features**
  * Added wallet endpoint to retrieve player currency holdings with multi-language names, descriptions, and optional icons
  * Returns balances per currency and preserves zero/very large amounts
  * API-first retrieval with background synchronization to keep DB in sync and DB fallback for availability
* **Tests**
  * Comprehensive test suites for the wallet endpoint, wallet service, and API key flows covering success, error, and edge cases

### CODE REVIEW

File: app/database/repositories/wallet_repository.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        self.session = session
+
+    async def get_by_id(self, _id: int) -> Optional[Wallet]:
+        raise NotImplementedError("Get by ID is not implemented for ApiKeys.")
+
+    async def get_all(self) -> list[Wallet]:
+        raise NotImplementedError("Get all is not implemented for ApiKeys.")
+
+    async def upsert(self, entity: Wallet) -> Wallet:
+        raise NotImplementedError("Upsert is not implemented for Wallet.")
```
[the remark this answers is not in the document]
Comment (JadeArkadian) [reply app/database/repositories/wallet_repository.py:26]: Done

File: app/services/wallet_service.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        """
+        try:
+            wallet_data = await self._get_wallet_from_api()
+            # Sync with DB in background
+            asyncio.create_task(self._sync_wallet_to_db(wallet_data, account_uuid))
+            return await self._build_response(wallet_data)
+
+        except Exception as e:
+            logger.warning(f"Failed to fetch wallet from GW2 API: {e}. Falling back to database.")
+            return await self._get_wallet_from_db(account_uuid)
```
[the remark this answers is not in the document]
Comment (JadeArkadian) [reply app/services/wallet_service.py:45]: Wont fix.... yet
